src/plugins/discourse/createGraph.js

```diff
@@ -0,0 +1,165 @@
+// @flow
+
+import {
+  Graph,
+  NodeAddress,
+  EdgeAddress,
+  type Node,
+  type Edge,
+  type NodeAddressT,
+} from "../../core/graph";
+import {type PostId, type TopicId, type Post, type Topic} from "./fetch";
+import {type DiscourseData} from "./mirror";
+import {
+  topicNodeType,
+  postNodeType,
+  userNodeType,
+  authorsPostEdgeType,
+  authorsTopicEdgeType,
+  postRepliesEdgeType,
+  topicContainsPostEdgeType,
+} from "./declaration";
+
+export function topicAddress(serverUrl: string, id: TopicId): NodeAddressT {
+  return NodeAddress.append(topicNodeType.prefix, serverUrl, String(id));
+}
+export function postAddress(serverUrl: string, id: PostId): NodeAddressT {
+  return NodeAddress.append(postNodeType.prefix, serverUrl, String(id));
+}
+export function userAddress(serverUrl: string, username: string): NodeAddressT {
+  return NodeAddress.append(userNodeType.prefix, serverUrl, username);
+}
+
+export function userNode(serverUrl: string, username: string): Node {
+  const url = `${serverUrl}/u/${username}/`;
```

Comment:
Presumably the server URL is trusted input; are there any restrictions
on the usernames, or should this be validated/sanitized?

Reply:
discourse usernames are only letters, underscores, and numbers. should be fine.